sematic/db/models/job.py

```diff
@@ -0,0 +1,163 @@
+    detail_serialization which can contain lots of information about the Kubernetes
+    job & related pod(s).
+
+    Attributes
+    ----------
+    name:
+        The k8s name of the job. Used instead of an 'id' field because
+        K8s identifies jobs by namespace and name.
+    namespace:
+        The k8s namespace of the job. In conjunction with name, forms
+        the unique id for the job.
+    run_id:
+        The id of the run the job is associated with.
+    last_updated_epoch_seconds:
+        The time the job status was last synced with Kubernetes. Pulled from
+        the most recent status to aid in queries.
+    state:
+        The simple state of the job. Pulled from the most recent status to
+        aid in queries.
+    kind:
+        Whether the job is for executing a run or a resolution.
+    message:
+        The human-readable message for the latest status. Pulled from
+        the most recent status to aid in queries.
+    detail_serialization:
+        A json-encoded serialization of detailed state information for the job
+    status_history_serializations:
+        A list of json-encoded serializations of job statuses
+    created_at:
+        The time the DB record for the job was created
+    updated_at:
+        The time the DB record for the job was last updated
+    """
+
+    __tablename__ = "jobs"
+
+    name: str = Column(types.String(), primary_key=True)
+    namespace: str = Column(types.String(), primary_key=True)
+    run_id: str = Column(types.String(), nullable=False)
+    last_updated_epoch_seconds: float = Column(types.Float(), nullable=False)
+    state: KubernetesJobStateString = Column(types.String(), nullable=False)
+    kind: JobKindString = Column(types.String(), nullable=False)
+    message: str = Column(types.String(), nullable=False)
+    detail_serialization: Dict[str, Any] = Column(  # type: ignore
+        types.String(), nullable=False
+    )
+    status_history_serializations: List[Dict[str, Union[str, float]]] = Column(
+        types.JSON(), nullable=False
+    )
+    created_at: datetime.datetime = Column(
+        types.DateTime(), nullable=False, default=datetime.datetime.utcnow
+    )
+    updated_at: datetime.datetime = Column(
+        types.DateTime(),
+        nullable=False,
+        default=datetime.datetime.utcnow,
+        onupdate=datetime.datetime.utcnow,
+    )
+
+    @classmethod
+    def new(
+        cls,
+        name: str,
+        namespace: str,
+        run_id: str,
+        status: JobStatus,
+        details: JobDetails,
+        kind: JobKindString,
+    ):
+        return Job(
+            name=name,
+            namespace=namespace,
+            run_id=run_id,
+            last_updated_epoch_seconds=status.last_updated_epoch_seconds,
+            state=status.state,
+            kind=kind,
+            message=status.message,
+            detail_serialization=asdict(details),
+            status_history_serializations=[asdict(status)],
+        )
```

Comment:
We have a `factories.py` module for this kind of stuff. This looks nice too, but I would favor consistency.

Reply:
changed